src/main/app/src/hooks/browseActivitiesFilter.js

```diff
@@ -0,0 +1,41 @@
+/** This hook filters the activities displayed on the browse page. */
+
+// This calls the appropriate filter based on activityType, or does nothing if no filter is set. 
+const filterActivities = (links, activityType, linkFilters, updateFilteredLinks) => {
+  if (Object.keys(linkFilters).length === 0)
+    updateFilteredLinks(links);
+  else
+    updateFilteredLinks(links.filter(link =>
+      activityType === "games" ? filterGames(link, linkFilters) :
+      activityType === "articles" ? filterArticles(link, linkFilters) :
+      filterVideos(link, linkFilters)
+    ));
+}
+
+/** 
+  This function takes in the game and filters, and checks if it satisfies the user entered number. 
+  This function is called using Javascript's filter function.
+*/
+const filterGames = (game, filters) => {
+  var input = parseInt(filters.numOfPlayers,10);
+    var min = parseInt(game.minPlayer,10);
+    var max = parseInt(game.maxPlayer,10);
+    
+    if( (min === 0 && max === 0) ||                 // If there are no restrictions on the number of players
```

Comment:
If you have the following if statement:
if (condition) {
  return true;
} else {
  return false;
}

It can be replaced by:
return condition